File: app/src/main/java/com/droidfeed/util/ObservableColorMatrix.java
package com.droidfeed.util;

import android.graphics.ColorMatrix;
import android.util.Property;

/**
 * Created by Dogan Gulcan on 11/7/17.
 */
public class ObservableColorMatrix extends ColorMatrix {

    private float saturation = 1f;

    public ObservableColorMatrix() {
        super();
    }

    public float getSaturation() {
        return saturation;
    }

    @Override
    public void setSaturation(float saturation) {
        this.saturation = saturation;
        super.setSaturation(saturation);
    }

    public static final Property<ObservableColorMatrix, Float> SATURATION = new FloatProperty<ObservableColorMatrix>("saturation") {

        @Override
        public void setValue(ObservableColorMatrix cm, float value) {
            cm.setSaturation(value);
        }

        @Override
        public Float get(ObservableColorMatrix cm) {
            return cm.getSaturation();
        }
    };

}

File: app/src/main/java/com/droidfeed/util/ObservableColorMatrix.kt
package com.droidfeed.util

import android.graphics.ColorMatrix
import android.util.Property

/**
 * Created by Dogan Gulcan on 11/5/17.
 */
class ObservableColorMatrix : ColorMatrix() {

    var saturation = 1f
        get() = field
        set(value) {
            field = value
        }

    companion object {

        val SATURATION: Property<ObservableColorMatrix, Float> by lazy {

            object : FloatProperty<ObservableColorMatrix>("saturation") {
                override fun setValue(cm: ObservableColorMatrix, value: Float) {
                    cm.saturation = value
                }

                override operator fun get(cm: ObservableColorMatrix): Float? {
                    return cm.saturation
                }
            }
        }
    }
}